scripts/xtensa-build-zephyr.py

```diff
@@ -329,7 +330,12 @@ def parse_args():
         │   └── sof-PLAT.ri
         └── sof-PLAT.ri\n
 				""")
-
+	tarball_args = parser.add_mutually_exclusive_group()
+	# argparse.BooleanOptionalAction requires Python 3.9
+	parser.set_defaults(tarball=True)
+	tarball_args.add_argument("--no-tarball", dest='tarball', action='store_false')
+	tarball_args.add_argument("--tarball", dest='tarball', action='store_true',
+			    help="""Create a tar file of the installed firmware files under build-sof-staging.""")
```

Comment:
if the default is true, then why do we need this?

Reply:
At least 2 reasons:
1. The test for reproducible builds should probably exclude tarballs but it does not right now (no symlinks on Windows right now). See above.
2. speed, disk usage, etc.

- 1 Should be easy enough to fix, there are already some exclusions.
- 2 is debatable.

Reply:
yea i am fine with `--no-tarball` but `--tarball` seems redundant

Reply:
It's redundant now but it can smooth a change of the default value in the future. This code is copied from 9d7c33adc where this happened.

Reply:
Dropped "--tarball". I was thinking this as an option for CI usage where CI infra that uses this would use --tarball explicitly, but no use yet.